### ISSUE TITLE
Gendox send Search Request to ProvenAI - Step 1

### ISSUE THREAD

**Issue by sekasx** [ctrl-space-labs/gendox-core#205]
### Pre-requisites: 
- Agent has been registered to ProvenAI and holds an agent ID VC
- Agent has authenticated to ProvenAI using its Agent ID VC and got an access token in return

### Description

Up until now, Gendox Agents was capable of performing semantic search only to the Documents belonging to the same project. This limitation was because there was no way to give permission to access data that didn't belong to the same person/organization.

With ProvenAI, the agents will be able to ask ProvenAI to give them extra content from other people, not associated with the same organization. 

When an Agent is registered to ProvenAI Gendox platform will perform in parallel:
- the existing semantic search using the method `embeddingService.findClosestSections`
- Search in ProvenAI platform using the ProvenAI-SDK

The result of the above will be merged into a single list of answers, and this concatenated list will be used as `Context` for the completion request, as it has already been implemented.

The search with ProvenAI-SDK is described in ctrl-space-labs/proven-ai#10
